src/Beacon.jsx

```diff
@@ -115,6 +124,9 @@ export class Beacon extends Component {
       // Update if inactive state or hash status have changed. Other state changes do not affect
       // the beacon view.
       return true;
+    } else if (this.state.tooltipBounds !== nextState.tooltipBounds && nextState.tooltipActive) {
+      // Update after getting values for arrow positioning
+      return true;
```

Comment:
Updating the component only when the tooltip should be activated to get the right values from DOM.